code/sound/speech.cpp

```diff
@@ -106,28 +107,40 @@ bool speech_play(const char *text)
 	}
 
 #ifdef _WIN32
-	size_t len = strlen(text);
-	wchar_t Conversion_buffer[MAX_SPEECH_CHAR_LEN];
-
-	if(len > (MAX_SPEECH_CHAR_LEN - 1)) {
-		len = MAX_SPEECH_CHAR_LEN - 1;
-	}
-
-	size_t count = 0;
-	for(size_t i = 0; i < len; i++) {
-		if(text[i] == '$') {
-			i++;
+	SCP_string work_buffer;
+
+	bool saw_dollar = false;
+	for (auto ch : unicode::codepoint_range(text)) {
+		if (ch == UNICODE_CHAR('$')) {
+			// Skip $ escape sequences which appear in briefing text
+			saw_dollar = false;
```

Comment:
...er... surely this should be `saw_dollar = true;`?

Reply:
also there should probably be a `continue` here as well

Reply:
Indeed.